a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -134,7 +134,7 @@ def execute(self, added_ports, loadbalancer, vthunder):
         """Execute get_info routine for a vThunder until it responds."""
         try:
             amphora_id = loadbalancer.amphorae[0].id
-            if len(added_ports[amphora_id]) > 0:
+            if added_ports and amphora_id in added_ports and len(added_ports[amphora_id]) > 0:
```

Comment:
Why the check amphora_id in added_ports is added here?
I think this check is not required here because if added_ports is there then it must be having the amphora_id in it as for each amp_id we define added_ports.
https://github.com/a10networks/a10-octavia/blob/master/a10_octavia/controller/worker/tasks/a10_network_tasks.py#L294